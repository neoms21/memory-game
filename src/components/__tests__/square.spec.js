import React from 'react';

import { shallow } from 'enzyme';
import Square from '../square';

describe('Square tests', () => {
  it('should render the square', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Square />);

    wrapper.simulate('click');

    expect(mockClick).not.toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire handleclick is present', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Square color="aa" handleClick={mockClick} />);

    wrapper.simulate('click');

    expect(mockClick).toHaveBeenCalledWith('aa');
  });
});
