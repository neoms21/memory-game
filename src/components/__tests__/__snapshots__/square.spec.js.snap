// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square tests should render the square 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "color": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  height: 70px;
  width: 70px;
  background: ",
          [Function],
          ";
  margin: 10px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  height: 70px;
  width: 70px;
  background: ",
            [Function],
            ";
  margin: 10px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
